code/script.js

```diff
@@ -1,19 +1,133 @@
-// Start here
+//Slows down the first alert to make everything on the site have time to load before the alert appears
+window.onload = function () { 
+  setTimeout(function() {
 
 // Step 1 - Welcome and introduction
-// Your code goes here
+//An alert with a welcoming message
 alert(
-  `Welcome to our Javascript Pizzeria. Ready to Start? - Click 'OK' to begin.`
+  `Welcome to our Pizzeria The Cheesy Slice! Ready to Start? - Click 'OK' to begin.`
+)
+
+//Using the prompt method to ask for the name of the customer and an alert uses that variable to greet the customer
+const guestName = prompt(`Dear guest, what is your name?`)
+console.log(guestName)
+
+alert(
+  `Hi ${guestName}! We will happily take your order!`
 )
 
 // Step 2 - Food choice
-// Your code goes here
+//Ask the customer for what type of food they want and confirm what they selected in an alert
+const foodChoice = parseInt(
+  prompt(
+    `What are you craving today?
+    1 - Pizza
+    2 - Pasta
+    3 - Salad
+    Please choose the number of what you would like to order.`
+  )
+)
+console.log(foodChoice)
+
+let selectedFood = ""
+
+if (foodChoice === 1) {
+selectedFood = "pizza"
+} else if (foodChoice === 2) {
+  selectedFood = "pasta"
+} else if (foodChoice === 3) {
+  selectedFood = "salad"
+} else {
+  alert(`We can't identify what type of food you want to order. Please choose a valid number.`)
+  process.exit(1)
+}
+
+alert(
+  `You have chosen to order a delicious ${selectedFood}!`
+)
 
 // Step 3 - Subtype choice
-// Your code goes here
+//Using the conditional statement switch method for the customer to choose what dish they want depending on what food they selected previously
+let subMenu = ""
+let dishName = ""
+let selectedDish = ""
+
+switch (selectedFood) {
+  case "pizza": 
+    subMenu = prompt(`What dish would you like to have?
+  1 - Vesuvio
+  2 - Capricciosa
+  3 - Veggie Delight
+  Please choose the number of what you would like to order.`)
+    dishName = ["Vesuvio", "Capricciosa", "Veggie Delight"]
+    break
+  case "pasta": 
+    subMenu = prompt(`What dish would you like to have?
+  1 - Bolognese
+  2 - Carbonara
+  3 - Frutti di Mare
+  Please choose the number of what you would like to order.`)
+    dishName = ["Bolognese", "Carbonara", "Frutti di Mare"]
+    break
+  case "salad": 
+    subMenu = prompt(`What dish would you like to have?
+  1 - Caesar salad
+  2 - Greek salad
+  3 - Cobb salad
+  Please choose the number of what you would like to order.`)
+    dishName = ["Caesar salad", "Greek salad", "Cobb salad"]
+    break
+  default:
+    alert(`We can't identify what dish you want to order. Please choose a valid number.`)
+    process.exit(1)
+}
+
+selectedDish = dishName[subMenu - 1] 
```

Comment:
As I was testing your program I noticed that typing in the wrong character (number OR letter) on the first choice (foodChoice) it correctly alerted me that something was wrong, but in the other part (selectedFood) I did get an error-message when typing in a random letter, but NOT when entering a higher number than 3. 

This may be because of the way dishName is declared on row 85. It will always take the number written in the prompt of subMenu and remove 1 from it. Hence, even if we type for example 8, it will try to find the selected dish from the array (7), but because it isn't there it will type in undefined in the alert window. The program will keep running, thinking the customer has chosen a type of dish, when in fact that dish doesn't exist :) 

Could you try the following method for making the program understand it can't go above the number 3?

```// Convert the input to a number
const chosenOption = parseInt(subMenu);

if (chosenOption >= 1 && chosenOption <= 3) {
  // Valid option selected
  selectedDish = dishName[chosenOption - 1];
  alert(`You have chosen to order a yummy ${selectedDish}!`);
} else {
  // Invalid option selected
  alert(`Invalid choice. Please choose a number between 1 and 3.`);
}```

Reply:
Thank you so much for discovering this and the suggestion of new code! I have used it and it seem to work! :)